viewing_party/party.py

```diff
@@ -1,21 +1,125 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict.update({"title": title})
+        movie_dict.update({"genre": genre})
+        movie_dict.update({"rating": rating})
+        return movie_dict
+    else:
+        return None   
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+    
+
+def watch_movie(user_data, title):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == title:
+            user_data["watched"].append(movie_dict)
+            user_data["watchlist"].remove(movie_dict)
+            return user_data
+    return user_data
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    rating_total = 0
+    
+    for movie in user_data["watched"]:
+        if len(user_data["watched"]) == 0:
+            movie["rating"] = 0.0
+        rating_list.append(movie['rating'])
+    
+    for rating in rating_list:
+        rating_total += rating
+    
+    if len(rating_list) == 0:
+        return 0.0
+    else:
+        final_rating = rating_total / len(rating_list)
+        return final_rating 

```

Comment:
Nice job handling the empty list case. Consider putting this check at the top as a guard clause in order to get it out of the way, which lets the main logic not be under a conditional.

```python
    if len(movies) < 1: # there are many ways we can check to see if movies is falsy
        return 0.0

    # remaining logic goes here
```